disttask/framework/dispatcher/dispatcher.go

```diff
@@ -470,16 +476,16 @@ func (d *dispatcher) GetAllSchedulerIDs(ctx context.Context, gTaskID int64) ([]s
 	}
 	ids := make([]string, 0, len(schedulerIDs))
 	for _, id := range schedulerIDs {
-		if ok := matchServerInfo(serverInfos, id); ok {
+		if ok := matchServerInfo(serverInfos, id, d.id); ok {
 			ids = append(ids, id)
 		}
 	}
 	return ids, nil
 }
 
-func matchServerInfo(serverInfos map[string]*infosync.ServerInfo, schedulerID string) bool {
+func matchServerInfo(serverInfos map[string]*infosync.ServerInfo, schedulerID string, id string) bool {
 	for _, serverInfo := range serverInfos {
-		serverID := disttaskutil.GenerateExecID(serverInfo.IP, serverInfo.Port)
+		serverID := disttaskutil.GenerateExecID(serverInfo.IP, serverInfo.Port, id)
```

Comment:
Are there compatibility issues if you need to add this?
In addition, it's best to push a new PR  or change the title, which feels like test-related PR.

Reply:
I will revert it